File: benchmarks/go/fib.go
package main

import (
	"fmt"
	"math/big"
	"os"
	"strconv"
	"time"
)

func fib(n int, verbose bool) *big.Int {
	a := big.NewInt(0)
	b := big.NewInt(1)

	for i := 0; i < n; i++ {
		if verbose {
			fmt.Printf("Step %d: %s\n", i, a.String())
		}
		tmp := new(big.Int).Set(a)
		a.Set(b)
		b.Add(tmp, b)
	}
	return a
}

func main() {
	if len(os.Args) < 2 {
		fmt.Println("Usage: fib <N> [--verbose]")
		return
	}
	n, _ := strconv.Atoi(os.Args[1])
	verbose := len(os.Args) > 2 && os.Args[2] == "--verbose"

	start := time.Now()
	result := fib(n, verbose)
	elapsed := float64(time.Since(start).Nanoseconds()) / 1e6

	fmt.Printf("[Fibonacci] Result = %s\n", result.String())
	fmt.Printf("[Fibonacci] Time   = %.3f ms\n", elapsed)
}

File: benchmarks/rust/src/bin/fib.rs
use std::env;
use std::time::Instant;
use num_bigint::BigUint;
use num_traits::{One, Zero};

fn fib(n: usize, verbose: bool) -> BigUint {
    let mut a = BigUint::zero();
    let mut b = BigUint::one();
    for i in 0..n {
        if verbose {
            println!("Step {}: {}", i, a);
        }
        let tmp = a.clone();
        a = b.clone();
        b = tmp + b;
    }
    a
}

fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() < 2 {
        eprintln!("Usage: fib <N> [--verbose]");
        return;
    }

    let n: usize = args[1].parse().unwrap_or(0);
    let verbose = args.len() > 2 && args[2] == "--verbose";

    let start = Instant::now();
    let result = fib(n, verbose);
    let elapsed = start.elapsed().as_secs_f64() * 1000.0;

    println!("[Fibonacci] Result = {}", result);
    println!("[Fibonacci] Time   = {:.3} ms", elapsed);
}
